clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/StoreRepushCommand.java

```diff
@@ -87,15 +114,108 @@ public List<String> getShellCmd() {
+        StoreResponse storeResponse = parentCtrlCli.getStore(s);
+        if (storeResponse.getStore().getHybridStoreConfig() != null) {
+          ret.setHybrid(true);
+        }
+
+        StoreHealthAuditResponse storeHealthInfo = parentCtrlCli.listStorePushInfo(s, false);
+        Map<String, RegionPushDetails> regionPushDetails = storeHealthInfo.getRegionPushDetails();
+        if (!regionPushDetails.containsKey(destFabric)) {
+          return ret.failWithResult(RepushViabilityInfo.Result.NO_FUTURE_VERSION);
+        }
+        String latestTimestamp = regionPushDetails.get(destFabric).getPushStartTimestamp();
+        LocalDateTime latestPushStartTime = LocalDateTime.parse(latestTimestamp, DateTimeFormatter.ISO_LOCAL_DATE_TIME);
+
+        if (latestPushStartTime.isAfter(timestamp)) {
+          return ret.failWithResult(RepushViabilityInfo.Result.TIMESTAMP_MISMATCH);
+        }
+
+        MultiStoreStatusResponse response = parentCtrlCli.getFutureVersions(clusterName, s);
+        // No future version status for target region.
+        if (!response.getStoreStatusMap().containsKey(destFabric)) {
+          return ret.succeedWithResult(RepushViabilityInfo.Result.SUCCESS);
+        }
+
+        int futureVersion = Integer.parseInt(response.getStoreStatusMap().get(destFabric));
+        // No ongoing offline pushes detected for target region.
+        if (futureVersion == Store.NON_EXISTING_VERSION) {
+          return ret.succeedWithResult(RepushViabilityInfo.Result.SUCCESS);
+        }
+        // Find ongoing pushes for this store, skip.
+        return ret.failWithResult(RepushViabilityInfo.Result.ONGOING_PUSH);
+      }
+    } catch (VeniceException e) {
+      return ret.failWithResult(RepushViabilityInfo.Result.EXCEPTION_THROWN);
+    }
   }
 
   @Override
   public void execute() {
+    RepushViabilityInfo repushViability = getRepushViability();
+    this.repushViabilityResult = repushViability.getResult();
+    StoreRepushCommand.Params repushParams = getParams();
+    ControllerClient cli = repushParams.getPCtrlCliWithoutCluster();
+    if (!repushViability.isViable()) {
+      completeCoreWorkWithError("failure: " + repushViability.getResult().toString());
+      return;
+    }
+    if (!repushViability.isHybrid()) {
+      try {
+        String clusterName = cli.discoverCluster(repushParams.getStore()).getCluster();
+        try (ControllerClient parentCtrlCli =
+            buildControllerClient(clusterName, repushParams.getUrl(), repushParams.getSSLFactory())) {
+          ControllerResponse prepareResponse = parentCtrlCli.prepareDataRecovery(
+              repushParams.getSourceFabric(),
+              repushParams.getDestFabric(),
+              repushParams.getStore(),
+              -1,
+              Optional.empty());
+          if (prepareResponse.isError()) {
+            completeCoreWorkWithError("failure: " + prepareResponse.getError());
```

Comment:
Don't we need an early return if error happens, we don't want to continue invoking the dataRecovery API, right? In fact, seems that we should always call return after `completeCoreWorkWithError`?